platform/linux-generic/odp_pool.c

```diff
@@ -1537,3 +1552,242 @@ int odp_buffer_is_valid(odp_buffer_t buf)
 
 	return 1;
 }
+
+/* No actual head pointer alignment requirement. Anyway, require even byte address. */
+#define MIN_HEAD_ALIGN 2
+
+int odp_pool_ext_capability(odp_pool_type_t type, odp_pool_ext_capability_t *capa)
+{
+	odp_pool_stats_opt_t supported_stats;
+
+	if (type != ODP_POOL_PACKET)
+		return -1;
+
+	supported_stats.all = 0;
+
+	memset(capa, 0, sizeof(odp_pool_ext_capability_t));
+
+	capa->type           = type;
+	capa->max_pools      = ODP_CONFIG_POOLS - 1;
+	capa->min_cache_size = 0;
+	capa->max_cache_size = CONFIG_POOL_CACHE_MAX_SIZE;
+	capa->stats.all      = supported_stats.all;
+
+	capa->pkt.max_num_buf         = _odp_pool_glb->config.pkt_max_num;
+	capa->pkt.max_buf_size        = MAX_SIZE;
+	capa->pkt.odp_header_size     = sizeof(odp_packet_hdr_t);
+	capa->pkt.odp_trailer_size    = 0;
+	capa->pkt.min_mem_align       = ODP_CACHE_LINE_SIZE;
+	capa->pkt.min_buf_align       = ODP_CACHE_LINE_SIZE;
+	capa->pkt.min_head_align      = MIN_HEAD_ALIGN;
+	capa->pkt.buf_size_aligned    = 0;
+	capa->pkt.max_headroom        = CONFIG_PACKET_HEADROOM;
+	capa->pkt.max_headroom_size   = CONFIG_PACKET_HEADROOM;
+	capa->pkt.max_segs_per_pkt    = PKT_MAX_SEGS;
+	capa->pkt.max_uarea_size      = MAX_SIZE;
+
+	return 0;
+}
+
+void odp_pool_ext_param_init(odp_pool_type_t type, odp_pool_ext_param_t *param)
+{
+	uint32_t default_cache_size = _odp_pool_glb->config.local_cache_size;
+
+	memset(param, 0, sizeof(odp_pool_ext_param_t));
+
+	if (type != ODP_POOL_PACKET)
+		return;
+
+	param->type         = ODP_POOL_PACKET;
+	param->cache_size   = default_cache_size;
+	param->pkt.headroom = CONFIG_PACKET_HEADROOM;
+}
+
+static int check_pool_ext_param(const odp_pool_ext_param_t *param)
+{
+	uint32_t head_offset = sizeof(odp_packet_hdr_t) + param->pkt.app_header_size;
+
+	if (param->type != ODP_POOL_PACKET)
+		return -1;
+
+	if (param->cache_size > CONFIG_POOL_CACHE_MAX_SIZE) {
```

Comment:
Could you check the parameters against `odp_pool_ext_capability_t` (like it is done in `check_params()`)? This would ease code maintenance if/when the capability values are modified.

Reply:
Fixed in v15